pkg/proc/stack.go

```diff
@@ -164,6 +203,103 @@ func (it *stackIterator) Next() bool {
 	return true
 }
 
+// switchStack will use the current frame to determine if it's time to
+// switch between the system stack and the goroutine stack or vice versa.
+func (it *stackIterator) switchStack() bool {
```

Comment:
nit: Could be named better, also checks for "top of stack".

Reply:
Suggestions for a better name?